### PR TITLE
refactor: Use same ios-core

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

File: Tuist/Package.resolved [hunk 1]
```diff
@@ -86,8 +86,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/Infomaniak/ios-core",
       "state" : {
-        "revision" : "33ce7bc356d9938078a4ec596a535390e25d8650",
-        "version" : "10.0.3"
+        "branch" : "mail",
+        "revision" : "8178fa508021e1d34deed765f545d4333d695a8e"
```
Comment (adrien-coye) [inline Tuist/Package.resolved:90]: Sure, and once this PR is about to be merged, we commit a new breaking version of `ios-core` I imagine ?

**Review by adrien-coye — COMMENTED** [review]
~~I think login is broken at this point.~~ Edit this branch has Tuist4, the login works fine.
Checked the code, looks good to me 👍

**Comment by adrien-coye** [comment]
checked "*_*" pattern, only thing that shows up is bundle names so I think it's all good.
Screenshot 2024-07-11 at 14 33 06

File: Tuist/Package.resolved (hunk 1 above)
Comment (PhilippeWeidmann) [reply Tuist/Package.resolved:90]: Exact 🎉

**Review by adrien-coye — APPROVED** [review]
All good to me

**Pull request merged**